lib/virt.py

```diff
@@ -431,10 +424,19 @@ def install(self, location=None, kickstart=None, rpmpack=None, disk_format='raw'
 
             # host the custom RPM on a HTTP server, as Anaconda needs a YUM repo
             # to pull packages from
-            srv = util.BackgroundHTTPServer(NETWORK_HOST, http_port)
+            srv = util.BackgroundHTTPServer(NETWORK_HOST, 0)
             srv.add_dir(repo, 'repo')
             stack.enter_context(srv)
```

Comment:
To be consistent with other `BackgroundHTTPServer` use cases, shouldn't it be like this?

```suggestion
            srv = util.BackgroundHTTPServer(NETWORK_HOST, 0)
            stack.enter_context(srv)
            srv.add_dir(repo, 'repo')
```

Reply:
No, it's better to add the file before starting the server if possible (to fully eliminate any chance of race conditions).

In the other cases, it's not possible.

Reply:
Why not? For example in `hardening/anaconda` you can add `remediation-ds.xml` outside of `srv` context and so `add_file` before context as well.

Reply:
Hmm, that's a good point - the anaconda use case could probably be done before starting the server.

I'll fix it and add `.start()` to the server, so that object initialization is more clearly decoupled from starting the server and we don't rely on the sneaky context manager behavior.